lib/scala/pkg/src/test/scala/org/enso/pkg/ConfigSpec.scala

```diff
@@ -90,4 +95,214 @@ class ConfigSpec
+          )
+        ),
+        extendedGroups = List(
+          ExtendedComponentGroup(
+            module = ModuleReference(
+              LibraryName("Standard", "Base"),
+              Some(ModuleName("Group 2"))
+            ),
+            color   = None,
+            icon    = None,
+            exports = List(Component("bax", None))
+          )
+        )
+      )
+      parsed.componentGroups shouldEqual expectedComponentGroups
+
+      val serialized = parsed.toYaml
+      serialized should include("""component-groups:
+                                  |  new:
+                                  |  - module: Group 1
+                                  |    color: '#C047AB'
+                                  |    icon: icon-name
+                                  |    exports:
+                                  |    - name: foo
+                                  |      shortcut: f
+                                  |    - bar
+                                  |  extends:
+                                  |  - module: Standard.Base.Group 2
+                                  |    exports:
+                                  |    - bax""".stripMargin)
+    }
+
+    "correctly de-serialize empty components" in {
+      val config =
+        """name: FooBar
+          |component-groups:
+          |""".stripMargin
+      val parsed = Config.fromYaml(config).get
+
+      parsed.componentGroups shouldEqual ComponentGroups.empty
+    }
+
+    "fail to de-serialize unknown keys in component groups" in {
+      val config =
+        """name: FooBar
+          |component-groups:
+          |  foo:
+          |  - Group 1:
+          |    exports:
+          |    - bax
+          |""".stripMargin
+
+      Config.fromYaml(config) match {
+        case Failure(f: DecodingFailure) =>
+          Show[DecodingFailure].show(f) should include(
+            "Invalid component group"
+          )
+        case unexpected =>
+          fail(s"Unexpected result: $unexpected")
+      }
+    }
+
+    "fail to de-serialize invalid extended modules" in {
+      val config =
+        """name: FooBar
+          |component-groups:
+          |  extends:
+          |  - Group 1:
+          |    exports:
+          |    - bax
+          |""".stripMargin
+
+      Config.fromYaml(config) match {
+        case Failure(f: DecodingFailure) =>
+          Show[DecodingFailure].show(f) should include(
+            "Failed to decode 'Group 1' as module reference"
+          )
+        case unexpected =>
+          fail(s"Unexpected result: $unexpected")
+      }
+    }
+
+    "correctly de-serialize shortcuts" in {
+      val config =
+        """name: FooBar
+          |component-groups:
+          |  new:
+          |  - Group 1:
+          |    exports:
+          |    - foo:
+          |      shortcut: f
+          |    - bar:
+          |      shortcut: fgh
+          |    - baz:
+          |      shortcut: 0
+          |    - quux:
+          |      shortcut:
```

Comment:
```suggestion
          |  - Group 1:
          |    exports:
          |    - foo:
          |      shortcut: f
          |    - bar:
          |      shortcut: fgh
          |    - baz:
          |      shortcut: 0
          |    - quux:
          |      shortcut:
          |    - hmmm
```

Maybe worth adding a key with no shortcut field to also verify that the shortcut is handled correctly in this case?